### PR TITLE
sync again

### PR DESCRIPTION
**Pull request by jessopb**
rename syncSubscribe to syncLoop, taking an optional param to disable interval behavior
add syncLoop(disable = true) to 'focus' event listener inside the "readyForSync" hook.
be sure to do full syncLoop when entering settings page
testing on bob

### CODE REVIEW

File: ui/component/app/view.jsx [hunk 1]
```diff
@@ -272,9 +279,17 @@ function App(props: Props) {
...
-      syncSubscribe();
+      syncLoop();
+      console.log('triggered');
     }
-  }, [readyForSync, hasVerifiedEmail, syncSubscribe]);
+    // @if TARGET='web'
+    window.addEventListener('focus', () => syncLoop(true));
+
+    return () => {
+      window.removeEventListener('focus', () => syncLoop(true));
```
Comment (seanyesmunt) [inline ui/component/app/view.jsx:288]: You need to pull this into it's own function so the event listener is removed properly

```
window.removeEventListener('focus', doSyncOnFocus)
```

File: ui/redux/actions/app.js [hunk 2]
```diff
@@ -517,6 +517,7 @@ export function doAnalyticsBuffer(uri, bufferData) {
 
 export function doAnalyticsTagSync() {
   return (dispatch, getState) => {
+    console.log('DATS')
```
Comment (DispatchCommit) [inline ui/redux/actions/app.js:520]: Probably could remove this (it's also the reason the build fails linting)

File: ui/redux/actions/sync.js [hunk 3]
```diff
@@ -103,24 +103,28 @@ export const doGetSyncDesktop = (cb?, password) => (dispatch, getState) => {
   });
 };
 
-export function doSyncSubscribe() {
+export function doSyncLoop(noInterval) {
+  console.log('doSyncLoop')
```
Comment (DispatchCommit) [inline ui/redux/actions/sync.js:107]: This also

File: ui/redux/actions/sync.js [hunk 4]
```diff
@@ -103,24 +103,28 @@ export const doGetSyncDesktop = (cb?, password) => (dispatch, getState) => {
...
-      syncTimer = setInterval(() => {
-        const state = getState();
-        const syncEnabled = makeSelectClientSetting(SETTINGS.ENABLE_SYNC)(state);
-        if (syncEnabled) {
-          dispatch(doGetSyncDesktop((error, hasNewData) => dispatch(doHandleSyncComplete(error, hasNewData))));
-          dispatch(doAnalyticsTagSync());
-        }
-      }, SYNC_INTERVAL);
+      if (!noInterval) {
+        console.log('interval')
```
Comment (DispatchCommit) [inline ui/redux/actions/sync.js:118]: And this console log as well

File: ui/redux/actions/app.js (hunk 2 above)
Comment (jessopb) [reply ui/redux/actions/app.js:520]: Yes, all of these were troubleshooting duplicate api calls due to another issue in the first file.

**Review by seanyesmunt — APPROVED** [review]

**Pull request merged**